src/StreamJsonRpc.Shared/JsonRpc.cs

```diff
@@ -362,8 +418,34 @@ public Task<Result> InvokeWithCancellationAsync<Result>(string targetName, IRead
         /// <exception cref="ObjectDisposedException">If this instance of <see cref="JsonRpc"/> has been disposed.</exception>
         public Task NotifyAsync(string targetName, params object[] arguments)
         {
+            // If somebody calls InvokeInternal<T>(id, "method", null), the null is not passed as an item in the array.
+            // Instead, the compiler thinks that the null is the array itself and it'll pass null directly.
+            // To account for this case, we check for null below.
+            arguments = arguments ?? new object[] { null };
```

Comment:
This caters to the way C# in particular works. We're hacking in functionality to reverse something the compiler did, which in general would make it more difficult to predict what will happen when you call a params method. So I'm not a fan of 'fixing up' here. Rather, if someone wants to pass in null as a single argument, they can always use the syntax `new object[] { null }`.
Or if we want to make it Just Work by default, the proper way is to add an overload of the NotifyAsync method that accepts exactly one argument. That way, when `null` is the one argument, C# will pick that overload instead of the params one.